RELEASE.md

```diff
@@ -41,40 +41,73 @@ The content of the release notes differs depending on the type of release, speci
-    - `VERSION=v0.6.0`
-1. Create an annotated tag
-    - `git tag -s -a $VERSION -m $VERSION`.
-1. Push the tag to the GitHub repository:
-   > NOTE: `upstream` should be the name of the remote pointing to `github.com/kubernetes-sigs/cluster-api-provider-openstack`
-    - `git push upstream $VERSION`
-
-   This will cause the image to be automatically built by CI and pushed to the staging repository. As this only builds
-   the image, it only takes a few minutes.
-   It also triggers the [release](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/.github/workflows/release.yaml) workflow which will generate release notes and artifacts, and create a draft release in GitHub.
-1. Follow the [image promotion process](https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/README.md#image-promoter) to promote the image from the staging repo to `registry.k8s.io/capi-openstack`.
-   The staging repository can be inspected at https://console.cloud.google.com/gcr/images/k8s-staging-capi-openstack/GLOBAL. Be
+
+1. Creating Release Notes
+   - Switch to the main branch: `git checkout main`
+   - Create a new branch for the release notes**:
+     `git checkout -b release-notes-X.Y.Z origin/main`
+   - Generate the release notes: `RELEASE_TAG=vX.Y.Z make generate-release-notes`
+   - Replace `vX.Y.Z` with the new release tag you're creating.
+   - This command generates the release notes here
+     `releasenotes/<RELEASE_TAG>.md` .
+
+1. Next step is to clean up the release note manually.
+   - If release is not an alpha or a beta or release candidate, check for duplicates,
+     reverts, and incorrect classifications of PRs, and whatever release
+     creation tagged to be manually checked.
+   - For any superseded PRs (like same dependency uplifted multiple times, or
+     commit revertion) that provide no value to the release, move them to
+     Superseded section. This way the changes are acknowledged to be part of the
+     release, but not overwhelming the important changes contained in the release.
+   - Commit your changes, push the new branch and create a pull request:
+     - The commit and PR title should be 🚀 Release v1.x.y:
+       -`git commit -S -s -m ":rocket: Release vX.Y.Z"`
+       -`git push -u origin release-notes-X.Y.Z`
+     - Important! The commit should only contain the release notes file, nothing
+     else, otherwise automation will not work.
+   - Ask maintainers and release team members to review your pull request.
+
+   Once PR is merged following GitHub actions are triggered:
+
+   - GitHub action `Create Release` runs following jobs
+     - GitHub job `push_release_tags` will create and push the tags. This action
+     will also create release branch if its missing and release is `rc` or minor.
+     - GitHub job `create draft release` creates draft release. Don't publish the
+     release until release tag is visible in. Running actions are visible on the
```

Comment:
We need to follow image promotion below so let's just say we should not publish yet.
```suggestion
     release yet. Running actions are visible on the
```